### ISSUE TITLE
assets return 403 with 3.7.1

### ISSUE THREAD

**Issue by joepc74** [stancl/tenancy#1143]
### Bug description

Tenancy assets return 403.
This code:
`<img src="{{ asset('logo.svg') }}" height="37" />`
Works with 3.7, and with 2.7.1 return 403 Forbidden.

With favicon same issue: 
<link rel="shortcut icon" type="image/x-icon" href="{{ asset('favicon.ico') }}">
Return 403.

### Steps to reproduce

Blade code: `<img src="{{ asset('logo.svg') }}" height="37" />`
Direct url: `http://localhost.test/tenancy/assets/logo.svg`
Image dir: `storage\tenantleon\app\public`

### Expected behavior

Show assets.

### Laravel version

8.2

### stancl/tenancy version

3.7.1

**Comment by stancl** [comment]
That would be related to this change: https://github.com/archtechx/tenancy/commit/4af70d302ffcf19306cc16009bd94162f3743ca7

What operating system are you on?

**Comment by joepc74** [comment]
> That would be related to this change: [4af70d3](https://github.com/archtechx/tenancy/commit/4af70d302ffcf19306cc16009bd94162f3743ca7)
> 
> What operating system are you on?

Windows 11 with Xampp

**Comment by stancl** [comment]
The change we made uses `realpath()` to convert paths strings to absolute paths which we then validate as subdirectories of the `storage_path()`.

The function also returns `false` if the **path doesn't exist**. I think that might be what's happening here, though in that case you'd likely get 404 before and get 403 now.

I tested how `realpath()` works on Windows and it seems to handle both `\` and `/` just fine, so I'm not sure what part of the change would make this behave incorrectly on Windows.

**Comment by joepc74** [comment]
> The change we made uses `realpath()` to convert paths strings to absolute paths which we then validate as subdirectories of the `storage_path()`.
> 
> The function also returns `false` if the **path doesn't exist**. I think that might be what's happening here, though in that case you'd likely get 404 before and get 403 now.
> 
> I tested how `realpath()` works on Windows and it seems to handle both `\` and `/` just fine, so I'm not sure what part of the change would make this behave incorrectly on Windows.

realpath("{$allowedRoot}/{$path}") ----> C:\xampp\htdocs\avata\storage\tenantleon\app\public\logo.svg
$allowedRoot ----> C:\xampp\htdocs\avata\storage/tenantleon\app/public

**Comment by stancl** [comment]
Ah, so it seems that the `startsWith()` fails since `realpath()` returns paths with `\` while `storage_path()` returns the path with `/`. Thanks for providing those strings, I'll try to push a fix today

**Comment by stancl** [comment]
If you want, you can test if making this change would fix the problem for you:
```diff
- startsWith($allowedRoot))
+ startsWith(realpath($allowedRoot)))
```

**Comment by joepc74** [comment]
Works.

**Comment by ibarrasantiago** [comment]
I have the same issue and that change fixes it.

**Comment by xahmedtaha** [comment]
Any ETA on when this fix will be pushed? Should I make a pull request? @stancl

**Comment by stancl** [comment]
Just to confirm — is everyone who's experiencing this on Windows?

**Comment by xahmedtaha** [comment]
Yes, with the exact same issue because of the different slashes  /, \

**Comment by stancl** [comment]
Cool, thanks for the feedback here. Working on a refactor that implements this in a better way. `realpath()` is a bit of a pain since it can return `false` when the path doesn't exist, so I'm adding more logic to handle every possible case, as well as test every possible case while always returning 404 (no more 403) to avoid leaking any information (e.g. the existence of a file outside the storage root).

**Comment by stancl** [comment]
Fixed in [v3.7.2](https://github.com/archtechx/tenancy/releases/tag/v3.7.2)
